engine_modules/market/tasks.py

```diff
@@ -1,8 +1,127 @@
 # -*- coding: utf-8 -*-
+from django.db.models import Max
+
 from engine.tasks import ResolutionTask
 from engine_modules.market.models import CorporationMarket
 
 
+class UpdateBubblesTask(ResolutionTask):
+	"""
+	Update the bubble value on the CorporationMarket objects
+	"""
+	RESOLUTION_ORDER = 500
+
+	def run(self, game):
+		# We can force the use of bubbles  using the force_bubbles flag
+
+		# I do NOT remember why the test was "not hasattr(game, 'force_bubbles')", I left it here just in case
+		# but the check == False is necessary. TODO: check if 'hasattr' has meaning (-> can a game not have an attribute 'force_bubbles' ?)
+		if game.disable_side_effects and (not hasattr(game, 'force_bubbles') or not game.force_bubbles):
+			return
+
+		corporation_markets = CorporationMarket.objects.filter(corporation__game=game, turn=game.current_turn).annotate(maxval=Max('value')).order_by('market__name', '-value')
```

Comment:
also, maybe you could add a final filter : market.value == 0 || market == maxvalue ?